client/buy.go

```diff
@@ -26,11 +26,7 @@ import (
 	"github.com/snapcore/snapd/store"
 )
 
-type BuyResult struct {
-	State string `json:"state"`
-}
-
-func (client *Client) Buy(opts *store.BuyOptions) (*BuyResult, error) {
+func (client *Client) Buy(opts *store.BuyOptions) (*store.BuyResult, error) {
```

Comment:
The cross over between client and store here is strange. These have different needs, and are talking to different backends. BuyOptions holds auth.UserState, for example.

Let's please have local BuyOptions and BuyResult types, and constrain them to the needs of the snapd API, which is what this package is really about.